### PR TITLE
Decode xlml

### PR DESCRIPTION
**Pull request by gobbleturk**
XLML test for decoder.py, which tests that decode.py works via counting number of decoded tokens

Also: 
* Fixes decoder.py by removing the config.use_pjrt
* Refactors write_metrics_locally to max_utils so it can be shared by both train.py and decoder.py
* Rename eval_assert.test_flops to eval_assert.assert_metric_average to indicate it be used more generally such as this test

### CODE REVIEW

File: end_to_end/test_tflops_v4_8_one_slice.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@ fi
 #Setup and Train
 bash setup.sh
 python3 MaxText/train.py MaxText/configs/base.yml run_name=$RUN_NAME\
-    steps=150 reuse_example_batch=1 remat_policy='full' enable_checkpointing=False metrics_file='metrics.txt'\
+    steps=15 reuse_example_batch=1 remat_policy='full' enable_checkpointing=False metrics_file='metrics.txt'\
```
Comment (priyanka-ganesha) [inline end_to_end/test_tflops_v4_8_one_slice.sh:20]: I think you may have missed changing this back to 150 after testing

File: end_to_end/test_decode_v4_8_one_slice.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+set -e
+
+NUM_TOKEN_THRESHOLD=${1}
+OUTPUT_PATH=${2}
```
Comment (priyanka-ganesha) [inline end_to_end/test_decode_v4_8_one_slice.sh:5]: We may need DATASET_PATH as well since base.yml has a default value for it

**Review by priyanka-ganesha — APPROVED** [review]

File: end_to_end/test_tflops_v4_8_one_slice.sh (hunk 1 above)
Comment (gobbleturk) [reply end_to_end/test_tflops_v4_8_one_slice.sh]: Fixed.

File: end_to_end/test_decode_v4_8_one_slice.sh (hunk 2 above)
Comment (gobbleturk) [reply end_to_end/test_decode_v4_8_one_slice.sh:5]: Good catch! I think decode.py doesn't use the training examples - but it does rely on the vocab/tokenizer so we do need to pass in the dataset path.

**Pull request merged**